lib/ui/bottombar/test_bottombar_navigation.dart

```diff
@@ -0,0 +1,180 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:farmsmart_flutter/data/bloc/article/ArticleListProvider.dart';
+import 'package:farmsmart_flutter/data/repositories/FlameLink.dart';
+import 'package:farmsmart_flutter/data/repositories/article/ArticleRepositoryInterface.dart';
+import 'package:farmsmart_flutter/data/repositories/article/implementation/ArticlesRepositoryFlamelink.dart';
+import 'package:farmsmart_flutter/ui/bottombar/tab_navigator.dart';
+import 'package:farmsmart_flutter/ui/discover/ArticleList.dart';
+import 'package:farmsmart_flutter/ui/playground/data/playground_datasource_impl.dart';
+import 'package:farmsmart_flutter/ui/playground/playground_view.dart';
+import 'package:farmsmart_flutter/utils/assets.dart';
+import 'package:flutter/material.dart';
+
+import 'bottom_navigation_bar_controller.dart';
+
+final cms =
+    FlameLink(store: Firestore.instance, environment: Environment.development);
+final articleRepo = ArticlesRepositoryFlameLink(cms);
+
+void main() => runApp(SampleApp());
+
+//TODO Just for testing purposes, should be removed before merge.
+class SampleApp extends StatelessWidget {
+
+  static final double bottomBarIconSize = 25;
+
+  static final Widget _discoverPage = ArticleList(
+      viewModelProvider: ArticleListProvider(
+          title: 'Discover',
+          repository: articleRepo,
+          group: ArticleCollectionGroup.discovery));
+
+  static final Widget _myPlotPage = FirstPage();
+  static final Widget _profitLossPage = SecondPage();
+  static final Widget _communityPage = SecondPage();
+
+  static final Widget _playgroundView = PlaygroundView(
+    widgetList: PlaygroundDataSourceImpl().getList(),
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: ThemeData(
+        fontFamily: 'GT-America-Standard',
+        brightness: Brightness.light,
+        scaffoldBackgroundColor: Color(0xFFFFFFFF),
+        primaryColor: const Color(0xFFFFFFFF),
+        accentColor: const Color(0xFF757575),
+      ),
+      home: BottomNavigationBarController(
+        tabs: <TabNavigator>[
+          TabNavigator(
+            child: _myPlotPage,
+            navigatorKey: GlobalKey<NavigatorState>(),
+            barItem: BottomNavigationBarItem(
+              activeIcon: Image.asset(
+                Assets.BOTTOM_BAR_MY_PLOT_SELECTED,
+                height: bottomBarIconSize,
+              ),
+              icon: Image.asset(
+                Assets.BOTTOM_BAR_MY_PLOT_UNSELECTED,
+                height: bottomBarIconSize,
+              ),
+              title: Text(
+                'My Plot',
+              ),
+            ),
+          ),
+          TabNavigator(
```

Comment:
I have left the `BottomNavigationBarItem` as a parameter because not always will be an Icon so you can add any widget (like the profile image) as a navigation bar item but I'm agree with you to create the nav key inside the `TabNavigator` widget to avoid passing it as a parameter because it's always the same. I will change that to create this `navigatorkey` inside the `TabNavigator`

Reply:
👍